SU2_CFD/src/CMarkerProfileReaderFVM.cpp

```diff
@@ -0,0 +1,482 @@
+ * SU2 International Developers Society <www.su2devsociety.org>
+ * with selected contributions from the open-source community.
+ *
+ * The main research teams contributing to the current release are:
+ *  - Prof. Juan J. Alonso's group at Stanford University.
+ *  - Prof. Piero Colonna's group at Delft University of Technology.
+ *  - Prof. Nicolas R. Gauger's group at Kaiserslautern University of Technology.
+ *  - Prof. Alberto Guardone's group at Polytechnic University of Milan.
+ *  - Prof. Rafael Palacios' group at Imperial College London.
+ *  - Prof. Vincent Terrapon's group at the University of Liege.
+ *  - Prof. Edwin van der Weide's group at the University of Twente.
+ *  - Lab. of New Concepts in Aeronautics at Tech. Institute of Aeronautics.
+ *
+ * Copyright 2012-2019, Francisco D. Palacios, Thomas D. Economon,
+ *                      Tim Albring, and the SU2 contributors.
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../include/CMarkerProfileReaderFVM.hpp"
+
+CMarkerProfileReaderFVM::CMarkerProfileReaderFVM(CGeometry      *val_geometry,
+                                                 CConfig        *val_config,
+                                                 string         val_filename,
+                                                 unsigned short val_kind_marker,
+                                                 unsigned short val_number_vars) {
+  
+  /*--- Store input values and pointers to class data. ---*/
+  
+  rank = SU2_MPI::GetRank();
+  size = SU2_MPI::GetSize();
+  
+  this->config    = val_config;
+  this->geometry  = val_geometry;
+  dimension       = geometry->GetnDim();
+  
+  filename     = val_filename;
+  markerType   = val_kind_marker;
+  numberOfVars = val_number_vars;
+  
+  /* Attempt to open the specified file. */
+  ifstream profile_file;
+  profile_file.open(filename.data(), ios::in);
+  
+  /* If the file is not found, then we merge the information necessary
+   and write a template marker profile file. Otherwise, we read and
+   store the information in the marker profile file. */
+  
+  if (profile_file.fail()) {
+    MergeProfileMarkers();
+    WriteMarkerProfileTemplate();
+  } else {
+    ReadMarkerProfile();
+  }
+  
+}
+
+CMarkerProfileReaderFVM::~CMarkerProfileReaderFVM(void) { }
+
+void CMarkerProfileReaderFVM::ReadMarkerProfile() {
```

Comment:
Again out of curiosity: Why don't you use the same file-structure as the Surface-Output uses? The inlet-files without column description can be a bit cryptic .

Reply:
As you have noticed, I have moved this into its own reader class, like the mesh readers, since I feel it fits more in line with input rather than output (the output is really just an extra feature to help with preparing your own file). 

Another issue is that the variables and in particular the markers that we need for the output are not readily available - we only process the markers for output that are found in MARKER_PLOTTING at the moment. We may change this in the near future, but for now, the cleanest way to implement the inlet profile files in my opinion was with the new class I have added. It still has a bug or two according to the regressions, but I am working on that. I left the format the same for now, but adding a variable header line could be interesting..

Reply:
> but adding a variable header line could be interesting..

Yep that would be great at some point.

Reply:
If you do not plan to add variable header lines in this PR (which would be fine for me) then you can mark this as resolved.